### ISSUE TITLE
API.add_note(..) silently fails

### ISSUE THREAD

**Issue by csaben** [marph91/joppy#23]
I have a script that has been sending me updated todos for the past few weeks. A few days ago the api object's 'add_note' stopped posting to what I can see on my side. 

Is this just me?

Thanks in advanced

**Comment by marph91** [comment]
Hi, could you post your Joplin version? I can take a look at this weekend probably.

Also, could you check that the parent notebook is set correctly? Maybe the todo is created, but in a random notebook.

**Comment by csaben** [comment]
My joplin version is 2.9.17. And I checked the notebook id twice but the issue persists.

Sorry for the delayed response!

**Comment by marph91** [comment]
I did a few tests with Joplin 2.9.17 and a slightly modified joppy:

```python
from joppy.api import Api

api = Api(token=YOUR_TOKEN)
notebook_id = api.add_notebook(title="A" * 10)

# Note with valid parent notebook id
api.add_note(title="A" * 10, body="A" * 10, parent_id=notebook_id)

# Note with invalid parent notebook id
invalid_notebook_id = notebook_id + "a"
api.add_note(title="A" * 10, body="A" * 10, parent_id=invalid_notebook_id)

# Check the notes
all_notes = api.get_all_notes()
print(len(all_notes))  # 2
all_notes_in_notebook = api.get_all_notes(notebook_id=notebook_id)
print(len(all_notes_in_notebook))  # 1
```

After both calls of `add_note`, notes are created and HTTP 200 is returned by Joplin. We can create a note via joppy that has an invalid parent notebook and is **not** displayed in the Joplin UI. However, it's visible in the API and probably somewhere inside Joplin. So it seems to be not an issue in joppy itself.

I guess that some argument in your script is valid in theory, but there is a Joplin constraint that is not propagated as error in the API. Could you maybe log some of these non-working requests and post the arguments here?

**Comment by csaben** [comment]
OK so I figured out the issue. The machine that I run my "joplin server" through where it read and writes via the API using my script had had the application open but never syncing. So I got home and loaded Joplin on said machine and got an explosion of new notes from testing the past few days remotely.

Notably, the ssh session that would post notes through said machine while trouble shooting remotely would say that I had successfully added notes and be able to see they were there but I just neglected to consider syncing wasn't happening from that machine.

TL;DR rookie mistake. 

Any suggestion for a way to do this such that when I add a new note it just writes directly to the cloud? That way I don't have to have joplin running on a machine syncing it.

Thanks for your quick help on this @marph91
